### DISCUSSION TITLE
Expose physics step system

### DISCUSSION THREAD

**Discussion by agg23** [jcornaz/heron#237] [Ideas]
It is useful to know when the physics system runs and act before/after it. There is a public enum `PhysicsSystem` that labels the various physics steps, but as far as I am aware, labels are local to the `SystemSet` they are contained in, so attempting to run a system `before/after()` one of the `PhysicsSystem` results in Bevy warning that the label is unknown.

Additionally, the physics system may tick only on certain frames (this is the default behavior), so it can be important to know whether the physics stepped. This case in particular can be covered by an event, but I'm not sure what the ideal solution is in general. I think an argument could be made that since physics is stepping in `PostUpdate`, it's unnecessary to directly run `before/after()` the step.

**Comment by jcornaz** [comment]
> but as far as I am aware, labels are local to the SystemSet they are contained in, so attempting to run a system before/after() one of the PhysicsSystem results in Bevy warning that the label is unknown.

No, they are not local to the system set, they are local to the stage. 

If you get this warning, it is because your system is not running in the same stage as the physics update (post-update). Therefore there is no problem because your system is already always before or after the physics update.

> the physics system may tick only on certain frames (this is the default behavior), so it can be important to know whether the physics stepped. 

the public system [`heron::should_run`](https://docs.rs/heron/2.3.0/heron/fn.should_run.html) tells you if the step should run for the current frame.

Note, you may also be interested in this issue #170 (and related PR #171)

**Reply by jcornaz** [reply]
You also have: [`PhysicsSteps::is_step_frame`](https://docs.rs/heron/2.3.0/heron/struct.PhysicsSteps.html#method.is_step_frame) available

**Reply by cswinter** [reply]
Related question, is there a way to control on which frame the physics system runs? Specifically, if I have a `SystemSet` with a `FixedTimestep` run criterion, can I configure the physics to run only on the frames where this `SystemSet` is is executed?

**Reply by jcornaz** [reply]
At the moment no. There has been an attempt to make this configurable here: #171.

This should definitely be configurable. But I don't have a lot of time to work on this at the moment.
